doc/design/feed_op_and_fectch_op.md

```diff
@@ -0,0 +1,120 @@
+# FeedOp and FetchOp Design Doc
+
+### Motivation
+
+Python programer needs an interface to feed the data to PaddlePaddle, run the model, and fetch the result from it. Since PaddlePaddle runtime only goes through a graph of ops, we need to design corresponding Ops and add them to the graph.
+
+### Challenge
+
+1. During the runtime of a particular Op, it only knows which `Variable` to be read from and written to. It doesn't have a direct access to python object.
+2. I/O involves copying data between python object and C++ object. In the current design, python only passes `ProgramDesc` to C++.
+
+### Solution
+
+To solve the first challenge, we add two **global** `Variable`s, `feed_result` and `fetch_result`. The operator can read/write these variables and later on it can be exchanged with python interface. The content of each is a `map<string, LoDTensor>`, where `string` is the `variable.name` and `LoDTensor` is the actual data. Be aware that the actually data is always on CPU.
+
+To solve the second challenge, we add two python interface `feed_value` and `fetch_value`. 
+
+```Python
+def feed_value(variable, np_variable):
+    """Overwrite feed_result[variable.name] with a numpy.array
+	
+    Args:
+    	variable:     Paddle Variable to be overwritten
+    	np_variable:  numpy_array to write the tensor feed_result[variable.name]
+    """
+
+def fetch_value(variable):
+    """Fetch the value of fetch_result[variable.name]
+       and convert it into a numpy.array
+    
+    Args:
+    	variable:     A Paddle Variable to be fetched
+    
+    Returns:
+    	numpy_array:  Value of the fetched variable
+    """
+```
+
+Now, we are able to implement the `FeedOp` and `FetchOp` as follows.
+
+```c++
+void FeedOp::Compute(const ExecutionContext& context)
+{
+  // Get Tensor reference in feed_result
+  string name = ctx.Output<Tensor>("Output")->name();
+  auto& var = GetScope()->GetVar("feed_result");
+  auto& input_tensor = var->Get<map<string, LoDTensor>>[name];
```

Comment:
I think we will need to access an Attribute here named "column", so could we know which column of the mini-batch should be copied into the `image` variable.